File: js/queue.js

//Adding or Pushing in Queue
let prompt = require("prompt-sync")({sigint:true});
let queueList = []
let inputNumbers =  prompt("How many elements you want to insert in the queue? = ")
while( queueList.length < inputNumbers ){
  ask = prompt("Enter the number you want to insert? = ");
  queueList.unshift(ask);
  
}

//Sorting Queue

let numbers= queueList;
sortNumbers = numbers.sort(function(a, b){return a - b})
console.log("The sorted queue would be" + " " + sortNumbers)

//Searching in Queue
var askValue = prompt("Enter your required value to search? = ")
if (queueList.includes(askValue)){
  console.log("Your searched value is existed in queue")
} else{
  console.log("Your searched Value is not  existed in queue")
}
//Removing in Queue
var removeValue = prompt("Enter the value which you want to remove? = ");
for ( var i=0 ; i< queueList.length;  i++){
    if (queueList[i] === removeValue ){
      queueList.splice(i,1);
      i--;
      console.log(queueList)
    }
  }

//Pop from Queue

queueList.pop(0)
console.log(queueList)

// insert in Queue
let position = prompt("Enter at which position you want to  insert the number? = ")
let insertNumber = prompt("Enter which number you want to insert? = ")
queueList.splice(position,0,insertNumber)
console.log(queueList)

File: Python/queue.py
class Queue:
    def __init__(self,queueElements):
        self.queueElements = queueElements

    def push(self):
        askElement = int(input("How  many elemnents you want to insert in queue?"))
        while len(self. queueElements) < askElement:
            addElement = int(input("Enter the elements you want to insert in queue?"))
            self.queueElements.append(addElement)
            print(self.queueElements)
    def sort(self):
        for  i in range(len(self.queueElements)):
            for j in range(i+1, len(self.queueElements)):
                if self.queueElements[i] > self.queueElements[j]:
                    self.queueElements[i] , self.queueElements[j] = self.queueElements[j] , self.queueElements[i]
        print(self.queueElements)
    def search(self):
        number = int(input('Enter the number you want to search? = '))
        if number in self.queueElements:

            print("Your searched number" ,number ,'is existed in queue.')
        else:
            print("The number you asked is not existed in the queue")

    def Pop(self):
            

            print(self.queueElements.pop(0))
            print(self.queueElements)

    def Insert(self):
        insertNumber = int(input("Enter the number you want to insert in the queue?"))
        definePosition = int(input("Define the position where you want to insert the number?"))
        if insertNumber != None and definePosition != None:
            self.queueElements.insert(definePosition,insertNumber)

            print(self.queueElements)
    
    



s = Queue([])
s.push()
s.sort()
s.search()
s.Pop()
s.Insert()